examples/plugin/callgraph/signatures.go

```diff
@@ -2,25 +2,40 @@ package main
 
 import (
 	_ "embed"
+	"fmt"
+	"log"
+	"os"
 
+	callgraphv1 "buf.build/gen/go/safedep/api/protocolbuffers/go/safedep/messages/code/callgraph/v1"
 	"github.com/safedep/code/plugin/callgraph"
-	"github.com/safedep/dry/log"
 	"gopkg.in/yaml.v3"
 )
 
 //go:embed signatures.yaml
 var signatureYAML []byte
 
 type signatureFile struct {
-	Version    string                `yaml:"version"`
-	Signatures []callgraph.Signature `yaml:"signatures"`
+	Version    string                  `yaml:"version"`
+	Signatures []callgraphv1.Signature `yaml:"signatures"`
 }
 
-var parsedSignatures signatureFile
+var ParsedSignatures []*callgraphv1.Signature
 
 func init() {
-	err := yaml.Unmarshal(signatureYAML, &parsedSignatures)
+	var parsedSignatureFile signatureFile
+	err := yaml.Unmarshal(signatureYAML, &parsedSignatureFile)
 	if err != nil {
 		log.Fatalf("Failed to parse signature YAML: %v", err)
 	}
+
+	ParsedSignatures = make([]*callgraphv1.Signature, len(parsedSignatureFile.Signatures))
+	for i := range parsedSignatureFile.Signatures {
+		ParsedSignatures[i] = &parsedSignatureFile.Signatures[i]
+	}
+
+	err = callgraph.ValidateSignatures(ParsedSignatures)
```

Comment:
This need not be called by a consumer (checked in `NewSignatureMatcher`)
However, exposed this so that the signature loader/fetcher can fail-fast by verifying loaded signatures like in this case